scripts/ci_build_and_test_in_container.sh

```diff
@@ -265,12 +267,22 @@ fi
 if [[ "${RUN_INTEGRATED_TESTS}" = true ]]; then
   # We split the process in two steps. First installing the environment, then running the tests.
   or_die ninja ats_environment
+  
   # The tests are not run using ninja (`ninja --verbose ats_run`) because it swallows the output while all the simulations are running.
   # We directly use the script instead...
   # Temporarily, we are not adding the `--failIfTestsFail` options to `geos_ats.sh`.
   # Therefore, `ats` will exit with error code 0, even if some tests fail.
   # Add `--failIfTestsFail` when you want `failIfTestsFail` to reflect the content of the tests.
   integratedTests/geos_ats.sh
+  
+  # Rebaseline and pack into an archive
+  # Note: Keep a copy of the run logs, so that the correct version will be packed
+  mv integratedTests/TestResults integratedTests/TestResults_backup
+  integratedTests/geos_ats.sh -a rebaselinefailed
+  mv integratedTests/TestResults integratedTests/TestResults_rebaseline
+  mv integratedTests/TestResults_backup integratedTests/TestResults 
+  integratedTests/geos_ats.sh -a pack_baselines --baselineArchiveName ${DATA_EXCHANGE_DIR}/baseline_${DATA_BASENAME_WE}.tar.gz --baselineCacheDirectory ${DATA_EXCHANGE_DIR}
+
```

Comment:
maybe I missed something but are we always doing this on the CI? Or how are we triggering the baselines update once we verify the diffs are legitimate?

Reply:
actually is this just packing them but not pushing them?

Reply:
I think that the pattern is to pack and upload references every time we run the integrated tests (storage does not cost too much and we have a cleaning policy, so it's more than all right.

Once the package is created, then it's pushed at another step of the process, directly from within a GHA step.
That prevents us from installing the `GCP` toolchain in our `docker` envs.
Also the same upload step is used for both integrated tests and `geos` bundles indistinctly.

Reply:
Okay, makes sense. But do we really need pack and push after every commit? I am not so much worried about space, more about keeping track about what s the latest valid version of the baselines. Like I have missed how we are keeping track of that.

Reply:
We only push when it's computed, so everybody's able to check the results.
I'd say that the latest valid version of the baseline will be defined in a `yaml` file (or equivalent), like today it's a git commit hash (to the `integratedTests` repo).

Reply:
yeah, well, people kind of set the label and then leave it so for some PR we will basically have a push for each commit but it's probably not that bad .

Reply:
Yes, this will probably happen (every time the `run integrated tests` label is there, at least).
I fine with it.